### PR TITLE
[CI] Use official Oracle JDK image instead of in-house one

### PR DESCRIPTION
**Pull request by Torch3333**
## Description

This replaces the usage of the in-house Oracle JDK image by the official Oracle images for jobs running on Ubuntu. 
We still need to use the in-house image for the Cosmos job because there is no official image for windows architecture.

## Related issues and/or PRs

- https://github.com/scalar-labs/scalardb/pull/1978

## Changes made

Update the CI to use the official Oracle JDK image for all Java LTS in jobs running on ubuntu.

## Checklist

> The following is a best-effort checklist. If any items in this checklist are not applicable to this PR or are dependent on other, unmerged PRs, please still mark the checkboxes after you have read and understood each item.

- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have updated the documentation to reflect the changes.
- [x] Any remaining open issues linked to this PR are documented and up-to-date (Jira, GitHub, etc.).
- [x] Tests (unit, integration, etc.) have been added for the changes.
- [x] My changes generate no new warnings.
- [x] Any dependent changes in other PRs have been merged and published.

## Additional notes (optional)

N/A

## Release notes

N/A

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -243,6 +240,10 @@ jobs:
   integration-test-for-cosmos:
     name: Cosmos DB integration test (${{ matrix.mode.label }})
     runs-on: windows-latest
+    env:
+      # This variable evaluates to: if {!(Temurin JDK 8) && !(Oracle JDK 8 or 11)} then {true} else {false}
+      SET_UP_INT_TEST_RUNTIME_JDK_WHEN_NOT_ORACLE_8_OR_11: "${{ (github.event_name == 'workflow_dispatch' && !(inputs.INT_TEST_JAVA_RUNTIME_VERSION == '8' && inputs.INT_TEST_JAVA_RUNTIME_VENDOR == 'temurin') && !(inputs.INT_TEST_JAVA_RUNTIME_VENDOR == 'oracle' && (inputs.INT_TEST_JAVA_RUNTIME_VERSION == '8' || inputs.INT_TEST_JAVA_RUNTIME_VERSION == '11'))) && 'true' || 'false' }}"
+      SET_UP_INT_TEST_RUNTIME_ORACLE_JDK_8_OR_11: "${{ (inputs.INT_TEST_JAVA_RUNTIME_VENDOR == 'oracle' && (inputs.INT_TEST_JAVA_RUNTIME_VERSION == '8' || inputs.INT_TEST_JAVA_RUNTIME_VERSION == '11')) &&  'true' || 'false' }}"
```
Comment (Torch3333) [inline .github/workflows/ci.yaml:246]: No official JDK image with windows architecture exists so we still need to use the in-house image.

**Review by komamitsu — APPROVED** [review]
LGTM! 👍

**Review by brfrn169 — APPROVED** [review]
LGTM! Thank you!

**Comment by brfrn169** [comment]
@Torch3333 Could you please fix the conflicts when you have time?

**Comment by Torch3333** [comment]
@brfrn169 
I fixed the conflicts.

**Pull request merged**